main.go

```diff
@@ -137,12 +137,21 @@ func getConfig() (*config, error) {
 
 func getCustomLoggerFromContext(ctx *gin.Context) *logrus.Entry {
 	deliveryID, ok := ctx.Get("delivery")
-	if !ok {
+	if !ok || !isStringType(deliveryID) {
 		return nil
```

Comment:
But wait, this cannot return `nil` - wasn't that the root cause of the crash? You should keep your first approach of `return logrus.WithField("delivery", "nil")`

I am sorry I didn't catch this in the previous round when it was removed.

Reply:
I understood your comment and it can and should be nil 🙃